can_validator/ids.py

```diff
@@ -0,0 +1,70 @@
+"""
+Generate ids.h file.
+Run this file to write just ids.h or main.py to write all files.
+"""
+import sys
+sys.path.append("ParseCAN")
+import ParseCAN
+from common import ids_path, spec_path
+
+# The VCU_MC_MESSAGE and MC_RESPONSE messages need to be combined into one enum, which will be stored here
+can_mc_reg_lines = []
+
+
+def write(output_path, spec_path):
+    """
+    Generate ids.h file, which includes enums for segments of type enum.
+
+    :param output_path: file to be written to
+    :param spec_path: CAN spec path
+    """
+    spec = ParseCAN.spec.can(spec_path)
+    with open(output_path, 'w') as f:
+        f.write('#ifndef _MY17_CAN_LIBRARY_IDS_H\n')
+        f.write('#define _MY17_CAN_LIBRARY_IDS_H\n\n')
+        f.write('#include "can_validator/fsae_can_spec.h"\n\n')
+
+        for message in spec.messages.values():
+            for segment_name, segment in message.segments.items():
+                if segment.c_type == 'enum':
+                    if not (message.name == "Vcu_Mc_Message" or message.name == "Mc_Response"):  # MC special cases
```

Comment:
Perhaps it'd be better to have a helper function called is_special_message() instead of checking (message.name == "Vcu_Mc_Message" or message.name == "Mc_Response"). This way, if you have to handle more special cases in the future, you only have to change that function
#ReadyForChange